core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -803,7 +807,7 @@ class BrokerServer(
       if (clientToControllerChannelManager != null)
         CoreUtils.swallow(clientToControllerChannelManager.shutdown(), this)
 
-      if (logManager != null)
+      if (logManager != null && lifecycleManager != null)
```

Comment:
Does that mean if there is an error instantiating `lifecycleManager` then we will also not shutdown `logManager`?

Reply:
Thanks @apoorvmittal10 

the body of the if 
```
      if (logManager != null && lifecycleManager != null)
        CoreUtils.swallow(logManager.shutdown(lifecycleManager.brokerEpoch), this)
```
so if `lifecycleManager` was null but `logManager` was not, there still would be an NPE.

having said that, 
the reason I encountered the NPE was because logManager creation failed, so logManager was null and the body of the if would have been skipped anyway.

Reply:
Yeah, I understand that. But there could be another case where `lifecycleManager` construction fails, in that case ideally we should close `logManager`, if instantiated, correct?

Reply:
Will it be safe to pass `-1` as broker epoch to `logManager.shutdown` as the broker is not compeletely instantiated?

Reply:
I also think this seems a bit odd. Having gone to all of the trouble to null-check references to lifecycleManager, I would have expected something more like:
```
if (logManager != null)
  CoreUtils.swallow(logManager.shutdown(if (lifecycleManager != null) lifecycleManager.brokerEpoch else -1, this)
```

Reply:
thanks, pushed a change